### DISCUSSION TITLE
Welcome to BCL Discussions!

### DISCUSSION THREAD

**Discussion by kflemin** [BuildingComponentLibrary/bcl-manifest#3] [General]
## Welcome!
  We’re using Discussions as a place to connect with other members of our community. You can:
  * Ask questions about the BCL
  * Share ideas
  * Engage with other community members

**Comment by Ski90Moo** [comment]
I am an engineer that uses the OpenStudio application.  Occasionally I write measures and quite often I create components and templates (for example, a space type template for the Australian National Construction Code Standard Reference building).  I am not a programmer but I, infrequently, use Github to help the programmers troubleshoot issues with the app.

I have read the BCL manifest and the https://bcl2.nrel.gov/contribute page.  All of this seems much more complicated than the previous BCL and at the moment, perhaps because I am not a dedicate programmer, I just do not understand the terminology.  I see this as a significant barrier to contributions.

In addition, I produce tutorial videos for people interest in learning the OpenStudio application.  I would like to create a video discussing how to upload content to the BCL, but, at the moment I am lost. 

Do I need to use the OpenStudio SDK for this?  Is there a resource, person, or organization that can help me with this?

**Reply by kflemin** [reply]
Hi, I'll email you directly!

**Reply by cberranger** [reply]
For what it's worth, it would be helpful to keep discussions open and public. I have have similar questions/comments as Ski90Moo and seeing responses taken away from here seems counter-productive.

**Reply by kflemin** [reply]
@cberranger, sure thing!  As a results of my discussion/tutorial with Ski90Moo, I have added additional resources to the contribute page (https://bcl2.nrel.gov/contribute) related to how to use GitHub.  I have also added more details on how to create a component repository as well as an example component (added to step1 of the contribute page).

Please feel free to add more discussions/questions here if you run into a specific issue. thanks!

**Reply by Ski90Moo** [reply]
I do intend to create a youtube tutorial for the process of contributing to the BCL.  After discussing this with Katherine, I realized the process is more complicated than the original BCL.  I still haven't had a chance to sit down and fully understand it, so unfortunately I have not created the tutorial yet.

**Reply by Ski90Moo** [reply]
@kflemin can you review the first part of this tutorial and let me know if I am on the right track? : https://1drv.ms/v/s!AseQmgZxffwzgrg-UuTKzPBSD695Jg?e=RZv7KO

**Reply by Ski90Moo** [reply]
@kflemin Do I need to periodically "Fetch Upstream"?  Or is this automatically pushed to me?
![2021-12-26_LI](https://user-images.githubusercontent.com/69771412/147416782-2189b165-a022-4d0b-8241-af9cb5979e87.jpg)

**Reply by kflemin** [reply]
@Ski90Moo, yes, you should fetch upstream if you want to make additional changes to the BCL Manifest. You'll want to make sure that you have the latest copy so that your changes can be merged cleanly/easily.

**Comment by virkanwal** [comment]
Hello, please may I know how can I get access to the BCL library for open studio measures. I have been asked to get the Auth key, but I'm not able to find a way on the new website. how to create my account.

**Reply by kflemin** [reply]
Hello @virkanwal, you no longer need an auth key in the OpenStudio Application to access measures.  Simply put 32 Xs or 32 characters and that should remove the prompt.  See this post for more info: https://unmethours.com/question/51404/bcl-account-registration/?answer=51408#post-id-51408

 This prompt will hopefully be removed in the next release of the app.

**Reply by virkanwal** [reply]
Hello, Thanks for the help. I tried using the 32 Xs or 32 characters, it didn't help. My OS version is 2.8. Kindly find attached the error I get while accessing the BCL library for measures.

![image](https://user-images.githubusercontent.com/92427026/137187563-a3b8493b-0270-4d26-96c9-8522a8c6461f.png)

**Reply by virkanwal** [reply]
@kflemin  Sorry for not tagging you earlier: I have a timeline for project submission and I'm stuck with this issue.

**Reply by kflemin** [reply]
@virkanwal are you able to set it in the OpenStudio App itself?

**Reply by kflemin** [reply]
can you send me the key you are trying to use?

**Reply by virkanwal** [reply]
@kflemin Yes I'm able to input in OS application.
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
![image](https://user-images.githubusercontent.com/92427026/137191245-69ebbb83-f634-4ea9-ac1e-82b9a57ae61a.png)

![image](https://user-images.githubusercontent.com/92427026/137190960-660f75b2-3673-4558-bc97-0c1383f14907.png)

**Comment by kflemin** [comment]
@virkanwal we have reconfigured a bunch of stuff on the server so that we can maintain backward compatibility all the way back to OS 2.8.  Can you try again? It should work now with the 32 Xs.

**Reply by virkanwal** [reply]
@kflemin Thank you so much. yes, now I'm able to access measures through the BCL using the 32 Xs key.

**Comment by Ski90Moo** [comment]
@kflemin I have set up a measures repository just like I set up my components repository, but it doesn't seem to be linked to the BCL.  Can you provide some guidance?

**Reply by kflemin** [reply]
@Ski90Moo, I don't see any releases on your measures repository...once you make a github release, the measures should be automatically added to BCL.

**Reply by Ski90Moo** [reply]
@kflemin I have made some releases, but I still do not see the measures on the BCL.

**Reply by kflemin** [reply]
@Ski90Moo, can you try a release with a more standard version?  something like: v0.0.1 or v0.1.0 or something like that, instead of just '0001'?

**Reply by Ski90Moo** [reply]
@kflemin Ok, done

**Reply by kflemin** [reply]
@Ski90Moo, I ended up adding your release manually...you can double check the BCL to make sure it all looks ok.  I'm not sure why the github action didn't trigger. Do you see error messages in "Recent Deliveries" under Settings -> Webhooks?

**Comment by Ski90Moo** [comment]
Ok, thank you Katherine.  I created a release with the measure in a zip
file.  Will this suffice?  Thanks, Mike

On Wed, Sep 21, 2022 at 7:45 PM Katherine Fleming ***@***.***>
wrote:

> @Ski90Moo <https://github.com/Ski90Moo>, I don't see any releases on your
> measures repository...once you make a github release, the measures should
> be automatically added to BCL.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/BuildingComponentLibrary/bcl-manifest/discussions/3#discussioncomment-3703207>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQUKBFARXD5YQHK562BXLJTV7NJSZANCNFSM4ZTSEREQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <BuildingComponentLibrary/bcl-manifest/repo-discussions/3/comments/3703207
> @github.com>
>


-- 
Best Regards,
Mike Lovejoy P.E.
Helix Energy Partners LLC
Helix, OR
http://www.helix-engineers.net/
<https://deref-mail.com/mail/client/aJLmMsp2k0Y/dereferrer/?redirectUrl=http%3A%2F%2Fwww.helix-engineers.net%2F>
P:541-379-0271

**Reply by kflemin** [reply]
There should be a release on this page: https://github.com/Ski90Moo/OpenStudio_Measures_HEPLLC/releases 
You should see a button that says "draft a new release" in the upper corner of the page. Make the release and click on the "publish release" green button at the bottom. That button triggers the indexing on BCL.  Also, you don't need to include the bcl_manifest.json file in your repo...it uses the file in the main bcl-manifest repo to add repos to BCL.  hope this helps!

**Comment by Ski90Moo** [comment]
No.  See attached screenshot.[image: 2022-11-07 (2).png]

On Mon, Nov 7, 2022 at 6:54 PM Katherine Fleming ***@***.***>
wrote:

> @Ski90Moo <https://github.com/Ski90Moo>, I ended up adding your release
> manually...you can double check the BCL to make sure it all looks ok. I'm
> not sure why the github action didn't trigger. Do you see error messages in
> "Recent Deliveries" under Settings -> Webhooks?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/BuildingComponentLibrary/bcl-manifest/discussions/3#discussioncomment-4079119>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQUKBFFDGQ6LFVUN2U7GQG3WHE64HANCNFSM4ZTSEREQ>
> .
> You are receiving this because you were mentioned.Message ID:
> <BuildingComponentLibrary/bcl-manifest/repo-discussions/3/comments/4079119
> @github.com>
>


-- 
Best Regards,
Mike Lovejoy P.E.
Helix Energy Partners LLC
Helix, OR
http://www.helix-engineers.net/
<https://deref-mail.com/mail/client/aJLmMsp2k0Y/dereferrer/?redirectUrl=http%3A%2F%2Fwww.helix-engineers.net%2F>
P:541-379-0271

**Comment by Ski90Moo** [comment]
See attached.[image: 2022-11-07 (3).png]

On Mon, Nov 7, 2022 at 9:46 PM Michael Lovejoy ***@***.***>
wrote:

> No.  See attached screenshot.[image: 2022-11-07 (2).png]
>
> On Mon, Nov 7, 2022 at 6:54 PM Katherine Fleming ***@***.***>
> wrote:
>
>> @Ski90Moo <https://github.com/Ski90Moo>, I ended up adding your release
>> manually...you can double check the BCL to make sure it all looks ok. I'm
>> not sure why the github action didn't trigger. Do you see error messages in
>> "Recent Deliveries" under Settings -> Webhooks?
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/BuildingComponentLibrary/bcl-manifest/discussions/3#discussioncomment-4079119>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AQUKBFFDGQ6LFVUN2U7GQG3WHE64HANCNFSM4ZTSEREQ>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> <BuildingComponentLibrary/bcl-manifest/repo-discussions/3/comments/4079119
>> @github.com>
>>
>
>
> --
> Best Regards,
> Mike Lovejoy P.E.
> Helix Energy Partners LLC
> Helix, OR
> http://www.helix-engineers.net/
> <https://deref-mail.com/mail/client/aJLmMsp2k0Y/dereferrer/?redirectUrl=http%3A%2F%2Fwww.helix-engineers.net%2F>
> P:541-379-0271
>


-- 
Best Regards,
Mike Lovejoy P.E.
Helix Energy Partners LLC
Helix, OR
http://www.helix-engineers.net/
<https://deref-mail.com/mail/client/aJLmMsp2k0Y/dereferrer/?redirectUrl=http%3A%2F%2Fwww.helix-engineers.net%2F>
P:541-379-0271
